modules/nextflow/src/main/groovy/nextflow/executor/HyperQueueExecutor.groovy

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2020-2022, Seqera Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package nextflow.executor
+
+import java.nio.file.Path
+
+import groovy.json.JsonSlurper
+import groovy.transform.CompileStatic
+import nextflow.processor.TaskRun
+import nextflow.util.ServiceName
+import nextflow.util.TupleHelper
+/**
+ * Implements executor for HyperQueue
+ *  https://github.com/It4innovations/hyperqueue/
+ *
+ * @author Paolo Di Tommaso <paolo.ditommaso@gmail.com>
+ * @author Henrik Nortamo <henrik.nortamo@csc.fi>
+ */
+@CompileStatic
+@ServiceName('hq')
+class HyperQueueExecutor extends AbstractGridExecutor {
+
+    private final JsonSlurper parser = new JsonSlurper()
+
+    @Override
+    protected String getHeaderToken() {
+        return '#HQ'
+    }
+
+    @Override
+    protected void register() {
+        super.register()
+        log.warn "The support for HyperQueue is an experimental feature and it may change in a future release"
+    }
+
+    @Override
+    protected List<String> getDirectives(TaskRun task, List<String> result) {
+
+        result << '--name' << getJobNameFor(task)
+        result << '--log' << quote(task.workDir.resolve(TaskRun.CMD_LOG))
+        result << '--cwd' << quote(task.workDir)
+
+        // No enforcement, Hq just makes sure that the allocated value is below the limit
+        if( task.config.getMemory() )
+            result << '--resource mem=' + (task.config.getMemory().toBytes()) << ''
```

Comment:
It would be a bit cleaner to use instead 

result << '--resource' << "mem=${task.config.getMemory().toBytes()}"

Same for the following ones